src/components/action-popover/__internal__/action-popover-utils.tsx

```diff
@@ -0,0 +1,39 @@
+import React from "react";
+import { ActionPopoverItem } from "../action-popover-item/action-popover-item.component";
+
+// Reusable type alias for item types
+type ReactItem = React.ReactChild | React.ReactFragment | React.ReactPortal;
+
+export const getItems = (
+  children: React.ReactNode | React.ReactNode[]
+): ReactItem[] =>
+  React.Children.toArray(children).filter(
+    (child) => React.isValidElement(child) && child.type === ActionPopoverItem
+  );
+
+export const isItemDisabled = (items: ReactItem[]) => (
```

Comment:
suggestion: we could simplify this and pass the item we want here instead of relying on currying to pass the items and then an index

```
const isItemDisabled = (item: ReactItem) => React.isValidElement(item) && !!item.props?.disabled